File: problemas/1100/1134.py
combustiveis = {
    1: 'Alcool',
    2: 'Gasolina',
    3: 'Diesel',
    4: 'Fim'
}

endKey = 0
abastecidos = dict()

for combustivel in combustiveis.items():
    if combustivel[1] == 'Fim':
        endKey = combustivel[0]
    abastecidos.update({combustivel[0]: 0})

while (tipo := int(input())) != endKey:
    if tipo in abastecidos:
        abastecidos[tipo] += 1

abastecidos.pop(endKey)
print('MUITO OBRIGADO')
for abastecido in abastecidos.items():
    print(f'{combustiveis[abastecido[0]]}: {abastecido[1]}')

File: problemas/1100/1134.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace uri1134
{
    internal static class Program
    {
        private static void Main()
        {
            var combustiveis = new Dictionary<int, string>
            {
                {1, "Alcool"},
                {2, "Gasolina"},
                {3, "Diesel"},
                {4, "Fim"}
            };

            int endKey = combustiveis.FirstOrDefault(x => x.Value == "Fim").Key;
            var abastecidos = new Dictionary<int, int>();
            int tipo;

            foreach (var combustivel in combustiveis)
            {
                abastecidos.Add(combustivel.Key, 0);
            }

            while ((tipo = int.Parse(Console.ReadLine())) != endKey)
            {
                if (abastecidos.ContainsKey(tipo))
                {
                    abastecidos[tipo]++;
                }
            }

            abastecidos.Remove(endKey);
            Console.WriteLine("MUITO OBRIGADO");
            foreach (var abastecido in abastecidos)
            {
                Console.WriteLine($"{combustiveis[abastecido.Key]}: {abastecido.Value}");
            }
        }
    }
}